File: quickSort.py
arr  = []
n = 0

def arrayInit():
    n = int(input('Nhap n :'))
    print(n)
    for i in range(n):
        arr.append(int(input('Nhap phan tu thu ' + str(i) + ' :')))
    return n


def showArray():
    for i in range(n):
        print('{} '.format(arr[i]))



def swap(arr,i,j):
    arr[i], arr[j] = arr[j] , arr[i]


def quickSort(l,r,arr):
    i = l
    j = r
    pivot = arr[r]
    while (i <= j):
        while (arr[i]) < pivot: i += 1
        while (arr[j] > pivot): j -= 1

        if (i <= j ):
            swap(arr,i,j)
            i+=1
            j-=1

    if (l < j) : quickSort(l , j , arr)
    if (i < r): quickSort(i , r, arr)



n = arrayInit()
quickSort(0,n-1,arr)
showArray()

File: QuickSort.cpp
#include <iostream>
#define capacity 100
using namespace std;
void arrayInit(int& n, int arr[]) {
	cin >> n;
	for (int i = 0; i < n; i++) {
		cin >> arr[i];
	}
}

void showArray(int& n, int arr[]) {
	for (int i = 0; i < n; i++) {
		cout << arr[i] << " ";
	}
}
void swap(int& a, int& b) {
	int temp = a;
	a = b;
	b = temp;
}

void quickSort(int arr[], int left, int right) {
	int i = left;
	int j = right;
	int pivot = arr[j];
	while (i <= j) {
		while (arr[i] < pivot) i++;
		while (arr[j] > pivot) j--;
		if (i <= j) {
			swap(arr[i], arr[j]);
			i++;
			j--;
		}
	}
	if (left < j) quickSort(arr, left, j);
	if (i < right) quickSort(arr, i, right);

}



int main() {
	int n, arr[capacity];
	arrayInit(n, arr);
	quickSort(arr, 0, n - 1);
	showArray(n, arr);
	return 0;
}

